### PR TITLE
apply mock-up repos for localization and restore test

### PR DESCRIPTION
**Pull request by OsmondJiang**
@superyyrrzz , you may want to change your git.yml spec test also.

### CODE REVIEW

File: test/docfx.Test/e2e/E2ETest.cs [hunk 1]
```diff
@@ -270,7 +274,7 @@ private static string ToSafePathString(string value)
 
                             foreach (var (path, content) in commit.Files)
                             {
-                                var blob = repo.ObjectDatabase.CreateBlob(new MemoryStream(Encoding.UTF8.GetBytes(content ?? "")));
+                                var blob = repo.ObjectDatabase.CreateBlob(new MemoryStream(Encoding.UTF8.GetBytes(content?.Replace("\r", "") ?? "")));
```
Comment (mingwli) [inline test/docfx.Test/e2e/E2ETest.cs]: >s(content?.Replace("\r", "") [](start = 112, length = 28)

why '\r' needs to be removed?

File: docs/specs/localization.yml [hunk 2]
```diff
@@ -373,10 +394,6 @@ outputs:
   xrefmap.json:
```
Comment (live1206) [inline docs/specs/localization.yml]: specify xrefmap.json content as well?

File: docs/specs/localization.yml [hunk 3]
```diff
@@ -2,92 +2,116 @@
 # [repository mapping] 
 commands:
   - restore --locale zh-cn
-  - build
-repo: https://github.com/docascode/docfx-test-dependencies#master
-restores:
-  git/github.com/docascode/docfx-test-dependencies.zh-cn/1dc2455be4a78b4fd29b5641e6fb8d687f37464a-master/dep.md: loc-DEP
```
Comment (yufeih) [inline docs/specs/localization.yml]: do you intend to remove all restores checks and the code in E2ETest?

Comment (OsmondJiang) [reply docs/specs/localization.yml]: yes, I used the build output to check the restores

---
In reply to: [231407212](https://github.com/dotnet/docfx/pull/3585#discussion_r231407212) [](ancestors = 231407212)

File: docs/specs/localization.yml [hunk 4]
```diff
@@ -111,13 +139,17 @@ outputs:
...
   - build --locale zh-cn
-repo: https://github.com/docascode/docfx-test-dependencies#loc-contribution
-inputs:
-  docfx.yml: |
-    contribution:
-      repository: https://github.com/dotnet/docfx.en-us
-    localization:
-      mapping: RepositoryAndFolder
+repos: 
+  https://docfx.com/abc.en-us#loc-contribution3:
```
Comment (yufeih) [inline docs/specs/localization.yml]: consider using `https://docs.com/`, it is a short valid link that is ours.

File: docs/specs/localization.yml [hunk 5]
```diff
@@ -2,92 +2,116 @@
...
-    }
+repos:
+  https://github.com/docascode/docfx-test-dependencies#share-restore-map:
+    - files:
+        docfx.yml: |
+          github:
+            userCache: github-user.json
+            resolveUsers: true
+          dependencies:
+            child1: https://github.com/docascode/docfx-test-dependencies.localization#master
```
Comment (yufeih) [inline docs/specs/localization.yml]: also consider making all mocked repos starting with `https://docs.com`, it is hard for me to see whether this dependency is a mocked repo or a real github repo. This helps me better understand our dependencies to https://github.com/docascode/docfx-test-dependencies.

File: test/docfx.Test/e2e/E2ETest.cs (hunk 1 above)
Comment (OsmondJiang) [reply test/docfx.Test/e2e/E2ETest.cs]: no need actually, will remove

---
In reply to: [231405628](https://github.com/dotnet/docfx/pull/3585#discussion_r231405628) [](ancestors = 231405628)

File: docs/specs/localization.yml (hunk 2 above)
Comment (OsmondJiang) [reply docs/specs/localization.yml]: good idea

---
In reply to: [231406503](https://github.com/dotnet/docfx/pull/3585#discussion_r231406503) [](ancestors = 231406503)

Comment (OsmondJiang) [reply docs/specs/localization.yml]: the below test has already covered, will remove it

---
In reply to: [231424602](https://github.com/dotnet/docfx/pull/3585#discussion_r231424602) [](ancestors = 231424602,231406503)

File: docs/specs/localization.yml (hunk 5 above)
Comment (OsmondJiang) [reply docs/specs/localization.yml]: some case need to be github.com/xxx to make it a github repo.

---
In reply to: [231411868](https://github.com/dotnet/docfx/pull/3585#discussion_r231411868) [](ancestors = 231411868)

File: test/docfx.Test/e2e/E2ETest.cs [hunk 6]
```diff
@@ -274,7 +274,7 @@ private static string ToSafePathString(string value)
 
                             foreach (var (path, content) in commit.Files)
                             {
-                                var blob = repo.ObjectDatabase.CreateBlob(new MemoryStream(Encoding.UTF8.GetBytes(content?.Replace("\r", "") ?? "")));
```
Comment (yufeih) [inline test/docfx.Test/e2e/E2ETest.cs]: we may still need this one if test content is multi-line and people have different line ending configs.

**Pull request merged**